File: src/main/java/io/FileInputReader.java
package io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class FileInputReader implements InputReader{

    private final String fileName;

    public FileInputReader(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public List<String> readAndGetData() {
        Path filePath = Paths.get(fileName);

        try {
            return Files.readAllLines(filePath);
        } catch (IOException e) {
            System.out.println(e);
        }

        return List.of();
    }
}

File: src/main/kotlin/inputReader/FileInputReader.kt
package inputReader

import java.nio.file.Paths
import kotlin.io.path.readBytes
import kotlin.io.path.readLines
import kotlin.io.path.readText

class FileInputReader(private val inputTextFileName: String): InputReader {
    override fun read(): String {
        val file = Paths.get(inputTextFileName)
        val text = file.readLines().joinToString("\n")
        if (text.isBlank())
            return ""
        return text
    }
}